### DISCUSSION TITLE
Bpi m2 Berry boot from sata drive?

### DISCUSSION THREAD

**Discussion by kiko81** [johang/sd-card-images#60] [General]
previously managed to boot bookworm image - see #58
But how to get used of sata port on board - want root there - [reference](https://forum.banana-pi.org/t/banana-pi-boot-your-rootfs-from-sata/924).
Did this with another image set - but there boot partition wasn't empty like boot partitions here
Combined that boot with bookworm's root on sata - booted, but kernel shown as in previous image.. and some apt packages fail since are kernel related
I aware repo `sd-images` but anyway..
Any suggestions?

**Comment by johang** [comment]
See #9.

**Comment by kiko81** [comment]
managed it.. in short:
dl-ed uboot from official debian daily images -> dd-ed to sd-> created 64mb partition and copied contents of root's boot folder there,
changed partuuid in boot.cmd to sata's root and removed the boot prefixes in first 2 lines
```
load ${devtype} ${devnum}:${distro_bootpart} ${kernel_addr_r} vmlinuz-6.1.0-9-armmp <- here removed boot perefix
load ${devtype} ${devnum}:${distro_bootpart} ${ramdisk_addr_r} initrd.uimg-6.1.0-9-armmp <- here removed boot perefix
# 3rd line removed at all
setenv bootargs root=PARTUUID="?????????" rw rootwait <- here changed partuuid to sata's root partition
for fdtpath in /lib/firmware/6.1.0-9-armmp/device-tree/${vendor} /lib/firmware/6.1.0-9-armmp/device-tree /usr/lib/linux-image-6.1.0-9-armmp/${vendor} /usr/lib/linux-image-6.1.0-9-armmp /usr/lib/linux/${vendor} /usr/lib/linux; do
  if test -e ${devtype} ${devnum}:${distro_bootpart} ${fdtpath}/${fdtfile}; then
    load ${devtype} ${devnum}:${distro_bootpart} ${fdt_addr_r} ${fdtpath}/${fdtfile}
    bootz ${kernel_addr_r} ${ramdisk_addr_r} ${fdt_addr_r}
  fi
done
bootz ${kernel_addr_r} ${ramdisk_addr_r} ${fdtcontroladdr}
```

made new boot.scr and viola
